api_client/python/timesketch_api_client/sigma.py

```diff
@@ -31,67 +32,267 @@ class Sigma(resource.BaseResource):
+        """Returns the rule falsepositives."""
+        if self._falsepositives:
+            return self._falsepositives
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('falsepositives', '')
+
+    @property
+    def author(self):
+        """Returns the rule author."""
+        if self._author:
+            return self._author
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('author', '')
+
+    @property
+    def date(self):
+        """Returns the rule date."""
+        if self._date:
+            return self._date
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('date', '')
+
+    @property
+    def modified(self):
+        """Returns the rule modified date."""
+        if self._modified:
+            return self._modified
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('modified', '')
+
+    @property
+    def logsource(self):
+        """Returns the rule logsource."""
+        if self._logsource:
+            return self._logsource
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('logsource', '')
+
+    @property
+    def detection(self):
+        """Returns the rule detection."""
+        if self._detection:
+            return self._detection
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('detection', '')
+
+    @property
+    def references(self):
+        """Returns the rule references."""
+        if self._references:
+            return self._references
+        sigma_data = self.data
+
+        if not sigma_data:
+            return ''
+
+        return sigma_data.get('references', '')
+
+
+    def from_rule_uuid(self, rule_uuid):
+        """Get a Sigma object from a rule uuid.
+
+        Args:
+            rule_uuid: Id of the sigma rule.
+
+        """
+        self._rule_uuid = rule_uuid
+        self.resource_uri = f'sigma/rule/{rule_uuid}'
+        super().__init__(
+            api=self.api, resource_uri=self.resource_uri)
+
+        self.lazyload_data(refresh_cache=True)
+
+    def from_text(self, rule_text):
+        """Get a Sigma object from a rule text.
+
+        Args:
+            rule_text: Rule text to be parsed.
+
+        """
+        self.resource_uri = '{0:s}/sigma/text/'.format(self.api.api_root)
+        data = {'title': 'Get_Sigma_by_text', 'content': rule_text}
+        response = self.api.session.post(self.resource_uri, data=data)
+        response_dict = error.get_response_json(response, logger)
+
+        objects = response_dict.get('objects')
+
+        if not objects:
+            logger.warning(
+                'Unable to parse rule with given text')
+
+        rule_dict = objects[0]
```

Comment:
I can see here a list of all these...

so you can re-use this in the `def from_manual` if that function simply accepts a dict... then you can do something like

```python
def _load_rule_dict(self, rule_dict):
   """.....
   """
   self._author = rule_dict.get('....
    ....
```

And then here you can simply call

```python
rule_dict = objects[0]
self._load_rule_dict(rule_dict)
````

and in the `from_manual`

```python
def from_manual(self, rule_dict):
  """"docstring....""""
  self._load_rule_dict(rule_dict)
```

or something like that....


also in here... you do `objects = response_dict.get('objects')` but you never check if `objects` exists, adn to verify it is a list that has at least one element....  a simple

```python
if objects:
  rule_dict = objects[0]
  self._load_rule_dict(rule_dict)
```

would at least be better... or
```python
if not objects:
  # throw an error, and/or log an error
rule_dict = objects[0]
self._load_rule_dict(rule_dict)
```

Reply:
Well I check just above, but I do not exit here, corect.

Reply:
ok, should be better